contracts/extensions/contracts/MilestonesExtension.sol

```diff
@@ -0,0 +1,218 @@
+    /// @dev Emits a 'MilestoneStatusChanged()' event.
+    /// @param _milestoneStatus New status of the milestone
+    function reviewMilestone(Status _milestoneStatus) external virtual {
+        _validateReviewMilestone(msg.sender, _milestoneStatus);
+        // Check if the milestone status is pending
+
+        milestones[upcomingMilestone].status = _milestoneStatus;
+
+        emit MilestoneStatusChanged(upcomingMilestone, _milestoneStatus);
+
+        if (_milestoneStatus == Status.Accepted) {
+            upcomingMilestone++;
+        }
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Check if sender is profile owner or member
+    /// @param _anchor Anchor of the profile
+    /// @param _sender The sender of the transaction
+    /// @return 'true' if the sender is the owner or member of the profile, otherwise 'false'
+    function _isProfileMember(address _anchor, address _sender) internal view virtual returns (bool) {
+        IRegistry registry = allo.getRegistry();
+        IRegistry.Profile memory profile = registry.getProfileByAnchor(_anchor);
+        return registry.isOwnerOrMemberOfProfile(profile.id, _sender);
+    }
+
+    /// @notice Sets the bid for a given `_bidderId` address, likely to match the recipient's address
+    /// @param _bidderId The address of the bidder
+    /// @param _proposalBid The amount that was bid
+    function _setProposalBid(address _bidderId, uint256 _proposalBid) internal virtual {
+        if (_proposalBid > maxBid) {
+            // If the proposal bid is greater than the max bid this will revert
+            revert MilestonesExtension_EXCEEDING_MAX_BID();
+        } else if (_proposalBid == 0) {
+            // If the proposal bid is 0, set it to the max bid
+            _proposalBid = maxBid;
+        }
+
+        bids[_bidderId] = _proposalBid;
+        emit SetBid(_bidderId, _proposalBid);
+    }
+
+    /// @notice Validates if the milestones can be set at this moment by the `_sender`
+    /// @param _sender The address setting the milestones
+    function _validateSetMilestones(address _sender) internal virtual {
+        _checkOnlyPoolManager(_sender);
+        if (milestones.length > 0) {
+            if (milestones[0].status != Status.None) revert MilestonesExtension_MILESTONES_ALREADY_SET();
+            delete milestones;
+        }
+    }
+
+    /// @notice Validates if the milestone can be submitted at this moment by the `_sender`
+    /// @param _recipientId The recipient id
+    /// @param _sender The address of the submitter
+    function _validateSubmitUpcomingMilestone(address _recipientId, address _sender) internal virtual {
+        // Check if the 'msg.sender' is accepted
+        if (!_isAcceptedRecipient(_recipientId)) revert MilestonesExtension_INVALID_RECIPIENT();
+        if (_sender != _recipientId && !_isProfileMember(_recipientId, _sender)) {
```

Comment:
I think it is safe to remove the connection with the registry, checking if _sender is equal to _recipientId is enough and we can remove the `_isProfileMember` since it is not being used elsewhere, wdyt?

Reply:
since we give the option to override `_isAcceptedRecipient` the developer can add this `isProfileMember` check there